src/velbustcp/lib/connection/tcp/network.py

```diff
@@ -59,39 +65,92 @@ def send(self, data: bytearray) -> None:
                 except Exception:
                     self.__logger.exception("Could not send data to client %s", client.address())
 
-    def __accept_sockets(self) -> None:
-        """Accepts clients from socket, if the tcp server is closed it will also close socket
-        """
+    def __get_bound_socket(self) -> socket.socket:
+        RETRY_DELAY = 5.0
 
-        while self.is_active():
+        while self.is_active() and not self.__bind_socket:
+            # First, try to initialize the SSL context (as we only need to do it once and once done
+            # we won’t need to return to trying this, most likely.)
+            if self.__options.ssl and not self.__context:
+                try:
+                    context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+                    context.load_cert_chain(self.__options.cert, keyfile=self.__options.pk)
+                    self.__context = context
+                except Exception as e:
+                    self.__logger.error("Could not initialize SSL for %s: %s", self.__options.address, e)
```

Comment:
Shorthand to automatically add stacktrace to the logging call. Check https://docs.python.org/3/library/logging.html#logging.Logger.exception

```suggestion
                    self.__logger.exception("Could not initialize SSL for %s", self.__options.address)
```